### ISSUE TITLE
Failing to Consume access token from Supabase Magic Link with useSupabaseClient<Database>();

### ISSUE THREAD

**Issue by GraniteEducation** [nuxt-modules/supabase#321]
### Version
@nuxtjs/supabase: 
nuxt: 

### Reproduction Link

Right now it seems that the Nuxt composable useSupabaseClient<Database>(); is struggling to consume my access token and redirect to set the session on the client. 

My work around was simply to NOT use the Nuxt composable useSupabaseClient<Database>(); and instead initialize my supabase client in the standard Typescript way. If you do this, it seems to consume the access token perfectly and you are good to go! 

I am going to reach out to the Nuxt team and see if they are aware of this issue (because I believe in the past it has worked). 

### Steps to reproduce
useSupabaseClient<Database>();  <-not working for me rn with magic link
 createClient(...) <-  IS working for me with magic link)

### What is Expected?

I would expect both 
useSupabaseClient<Database>();  &  createClient(...) to be able to consume the access token from the URL

### What is actually happening?
Only createClient(...) can consume access token. useSupabaseClient<Database>(); fails to redirect and consume access token.

**Comment by liamcharmer** [comment]
yes! I've had this same issue. I've reverted back to version ``0.3.5`` as that was working in a previous project I Used. 

@GraniteEducation Did you get any response in the discords?

**Comment by erados** [comment]
I’m encountering the same issue. Even worse, the session created by createClient isn’t compatible with useSupabaseClient, at least based on my experience.

It seems like this issue has been unresolved for about a year.
